test/bias-2015/test_benign_classifiers.py

```diff
@@ -0,0 +1,453 @@
+    variant.gnomad = None
+    score, bs1 = benign_classifiers.get_bs1(variant)
+    expected_score = 0
+    expected_bs1 = ""
+    assert (score, bs1) == (expected_score, expected_bs1)
+
+def test_get_bp1():
+    """
+    Test the get_bp1 function
+    """
+    # Variant is a missense variant in a gene primarily associated with truncating variants
+    truncating_genes = {"GENE1"}
+    variant = type('Variant', (object,), {})()
+    variant.geneName = "GENE1"
+    variant.consequence = "missense_variant"
+    score, bp1 = benign_classifiers.get_bp1(variant, truncating_genes)
+    expected_score = 1
+    expected_bp1 = "BP1: Missense variant type missense_variant in gene GENE1 which has over 80% truncating pathogenic variants"
+    assert (score, bp1) == (expected_score, expected_bp1)
+
+    # Variant is a truncating variant in a gene primarily associated with truncating variants
+    truncating_genes = {"GENE1"}
+    variant = type('Variant', (object,), {})()
+    variant.geneName = "GENE1"
+    variant.consequence = "stop_gained"
+    score, bp1 = benign_classifiers.get_bp1(variant, truncating_genes)
+    expected_score = 0
+    expected_bp1 = ""
+    assert (score, bp1) == (expected_score, expected_bp1)
+
+    # Variant is a missense variant in a gene not associated with truncating variants
+    truncating_genes = {"GENE2"}
+    variant = type('Variant', (object,), {})()
+    variant.geneName = "GENE1"
+    variant.consequence = "missense_variant"
+    score, bp1 = benign_classifiers.get_bp1(variant, truncating_genes)
+    expected_score = 0
+    expected_bp1 = ""
+    assert (score, bp1) == (expected_score, expected_bp1)
+
+    # Variant is a missense variant, but geneName is None
+    truncating_genes = {"GENE1"}
+    variant = type('Variant', (object,), {})()
+    variant.geneName = None
+    variant.consequence = "missense_variant"
+    score, bp1 = benign_classifiers.get_bp1(variant, truncating_genes)
+    expected_score = 0
+    expected_bp1 = ""
+    assert (score, bp1) == (expected_score, expected_bp1)
+
+    # Variant is a missense variant, but consequence is None
+    truncating_genes = {"GENE1"}
+    variant = type('Variant', (object,), {})()
+    variant.geneName = "GENE1"
+    variant.consequence = None
+    score, bp1 = benign_classifiers.get_bp1(variant, truncating_genes)
+    expected_score = 0
+    expected_bp1 = ""
+    assert (score, bp1) == (expected_score, expected_bp1)
+
+def test_get_bp3():
+    """
+    Test the get_bp3 function.
+    """
+
+    # Define the repeat regions
+    chrom_to_repeat_regions = {
+        "chr1": [(1000, 2000, "repeat_region")]
+    }
+
+    # In-frame insertion in a repeat region
+    variant = type('Variant', (object,), {})()
+    variant.chromosome = "chr1"
+    variant.position = 1500
+    variant.refAllele = "A"
+    variant.altAllele = "AAT"
+    variant.consequence = "inframe_insertion"
+    score, bp3 = benign_classifiers.get_bp3(variant, chrom_to_repeat_regions)
```

Comment:
Im not sure if this will pass as written. I'd look at the get_bp3 function and then reverse engineer the inputs.  

This classifier determines the variant length and uses that to identify if a variant is in or out of frame (doesn't go by consequence like the pathogenic one does).

Reply:
Indeed it is failing when I try to run it

`________________________________________________________________________________ test_get_bp3 _________________________________________________________________________________

    def test_get_bp3():
        """
        Test the get_bp3 function.
        """
    
        # Define the repeat regions
        chrom_to_repeat_regions = {
            "chr1": [(1000, 2000, "repeat_region")]
        }
    
        # In-frame insertion in a repeat region
        variant = type('Variant', (object,), {})()
        variant.chromosome = "chr1"
        variant.position = 1500
        variant.refAllele = "A"
        variant.altAllele = "AAT"
        variant.consequence = "inframe_insertion"
        score, bp3 = benign_classifiers.get_bp3(variant, chrom_to_repeat_regions)
        expected_score = 1
        expected_bp3 = "BP3 (1): In-frame INDEL of length 2 in repeat region chr1 1000-2000"
>       assert (score, bp3) == (expected_score, expected_bp3)
E       AssertionError: assert (0, '') == (1, 'BP3 (1):...r1 1000-2000')
E         At index 0 diff: 0 != 1
E         Use -v to get more diff

test/bias-2015/test_benign_classifiers.py:211: AssertionError`

Reply:
I am not sure why the refAllele and the altAllele is not passing. How should I rewrite it so it passes?

Reply:
Its looking for a difference in length that is divisible by three. Try adding an extra allele to your alternative allele, then it will be dropping 3 bases (a full codon) and you should get a different readout. 

Ex.
  variant.altAllele = "AAAT"